pkg/gitops/github.go

```diff
@@ -83,32 +55,51 @@ func (gh github) OpenPullRequest(ctx context.Context, p openPullRequestParams) (
 		Head:  gogh.String(p.head),
 		Base:  gogh.String(p.base),
 	}
-	pr, _, err := gh.client.PullRequests.Create(ctx, gh.owner, gh.repoName, req)
+	pr, _, err := gh.client.PullRequests.Create(ctx, gh.repo.owner, gh.repo.name, req)
 	if err != nil {
 		return "", fmt.Errorf("create: %w", err)
 	}
 	return *pr.HTMLURL, nil
 }
 
-func githubOwnerRepo(s string) (string, string, error) {
+// githubRepo represents a Github repository.
+type githubRepo struct {
+	url   stepconf.Secret
+	token stepconf.Secret
+	owner string
+	name  string
+}
+
+// NewGithubRepo return a new Github repository.
+func NewGithubRepo(url, user string, token stepconf.Secret) (*githubRepo, error) {
 	// Trim prefix.
-	prefix := "git@github.com:"
-	if !strings.HasPrefix(s, prefix) {
-		return "", "", fmt.Errorf("must start with %q", prefix)
+	prefix := "https://github.com/"
+	if !strings.HasPrefix(url, prefix) {
+		return nil, fmt.Errorf("must start with %q", prefix)
 	}
-	s = strings.TrimPrefix(s, prefix)
+	url = strings.TrimPrefix(url, prefix)
 
 	// Trim suffix.
 	suffix := ".git"
-	if !strings.HasSuffix(s, suffix) {
-		return "", "", fmt.Errorf("must end with %q", suffix)
+	if !strings.HasSuffix(url, suffix) {
+		return nil, fmt.Errorf("must end with %q", suffix)
 	}
-	s = strings.TrimSuffix(s, suffix)
+	url = strings.TrimSuffix(url, suffix)
 
 	// Split remaining URL for owner and repository name.
-	a := strings.Split(s, "/")
+	a := strings.Split(url, "/")
 	if len(a) != 2 {
-		return "", "", fmt.Errorf("must separate owner from repo with one /")
+		return nil, fmt.Errorf("must separate owner from repo with one /")
 	}
-	return a[0], a[1], nil
+
+	// Construct an authenticated HTTPS URL from plain URL and credentials.
+	authURL := fmt.Sprintf(
+		"https://%s:%s@github.com/%s/%s.git", user, string(token), a[0], a[1])
```

Comment:
is this a safe option to put the PAT into `https` link? 🤔  can't we maybe use a header, or is this an only option?

Reply:
if you want to use it in git there is also an option to put the PAT in the configs but I've never used that way before. (in git you can set up basic authentication out of the box)

Reply:
When you pass `username:password` before `@` inside an URL, it will actually be transformed to HTTP Basic Auth, won't be part of the URL.
![image](https://user-images.githubusercontent.com/74185105/109385241-bc35e700-78f2-11eb-9d1e-54d380ec7092.png)